src/cljs/andel/core.cljs

```diff
@@ -229,11 +206,40 @@
           [0 :infinity]
           [0 (- to line-start-offset)])
         (and (<= line-start-offset from) (<= from line-end-offset))
-        [(- from line-start-offset) (if (< to line-end-offset)
+        [(- from line-start-offset) (if (<= to line-end-offset)
                                       (- to line-start-offset)
                                       :infinity)]
         :else nil))
 
+(defn on-mouse-action! [[line col] selection?]
+  (swap-editor! state #(contr/set-caret-line-col % line col selection?)))
+
+(defn scroll [viewport-fn]
+  (let [once (atom true)]
+    (fn []
+      [:div {:style {:display :flex
+                     :flex "1"
+                     :overflow :hidden}
+             :ref (fn [e]
+                    (when e
+                      (swap-editor! state #(assoc-in % [:viewport :view-size] [(.-clientWidth e) (.-clientHeight e)])) 100)
```

Comment:
what I see is scroll fn uses global variable 'state when it actually needs just a viewport (model) and a callback for changing it as props.
(defn scroll [viewport on-change view]) where view could be a hiccup itself and not a special callback fn (like [editor-viewport state])